mobile/src/stateMachine/views/ViewAffirmation.tsx

```diff
@@ -1,71 +1,106 @@
 import { ViewState } from './base';
 import React from 'react';
-import { NotificationRequest } from 'expo-notifications';
 import { observer } from 'mobx-react';
-import { StyleSheet, Text } from 'react-native';
+import { StyleSheet, Text, Image } from 'react-native';
 import { MasloPage, Container, Button } from 'src/components';
 import { ScenarioTriggers } from '../abstractions';
 import AppController from 'src/controllers';
 import { observable } from 'mobx';
+import { Affirmation } from 'src/constants/QoL';
 
+const affirmationSources = [
+  require('../../assets/gif/affirmation1.gif'),
+  require('../../assets/gif/affirmation2.gif'),
+  require('../../assets/gif/affirmation3.gif'),
+  require('../../assets/gif/affirmation4.gif'),
+  require('../../assets/gif/affirmation5.gif'),
+  require('../../assets/gif/affirmation6.gif'),
+];
 @observer
 export class ViewAffirmationView extends ViewState {
 
   private get notificationsController() {
     return AppController.Instance.User.notifications;
   }
   @observable
-  private currentNotification: NotificationRequest = this.notificationsController.openedNotification;
+  public currentAffirmation: Affirmation;
 
   constructor(props) {
     super(props);
     this._contentHeight = this.persona.setupContainerHeightForceScroll();
   }
 
   async start() {
-    this.currentNotification = this.notificationsController.openedNotification;
+    // When opening this screen from a tap on the notification, this method is called before the Firebase network connection has been enabled, we add a loop and delay here to wait
+    for (let i = 0; i < 120; i++) {
+      if (this.currentAffirmation == null) {
+        this.currentAffirmation = await this.notificationsController.getAffirmationToShow();
+      } else {
+        break;
+      }
+      this.sleep(25);
```

Comment:
Where do you get this 25 value from? Why is it not more or less?

Reply:
120*25ms = 3000ms. 3 seconds may be too short